### DISCUSSION TITLE
Can't enable giant mushrooms in config

### DISCUSSION THREAD

**Discussion by Some-Doofus** [RakambdaOrg/FallingTree#347] [Q&A]
### Describe the bug

I want to make giant mushrooms able to be cut down as trees, but adding the mushroom stem block (minecraft:mushroom_stem) to the allowed logs and the mushroom blocks (minecraft:red_mushroom_block and minecraft:brown_mushroom_block) to the allowed leaves gives an error when I try to cut a mushroom stem block. The error reads: "Tree detection stopped, reason: Found block, fr.raksrinana.fallingtree.fabric.common.wrapper.BlockWrappers@46590dd0 that isn't allowed in the adjacent blocks." I'm not sure what this means exactly but I tried adding all the surrounding blocks to the allowed adjacent blocks list including mycelium, dirt, and the mushroom stem and mushroom blocks, but even cutting one stem block not adjacent to any block gives the error. Any info on how to fix this would be great, or if it's just a bug, well, good luck fixing it.

Cheers

### Minecraft version

1.18.2

### Fabric API version

0.50.0

### Mod version

3.5.0

**Comment by Rakambda** [comment]
Adjacent blocks are for a totally different thing; they allow you to define what blocks are allowed to be next to tree parts.

What you want to do is probably something like (I'm on mobile for now so I won't have the exact lock names):

* Add the "trunk" of the mushroom (minecraft:stem_block I guess) to the allowed logs [trees.allowedLogs]
* Add the mushroom "cap" to the allowed leaves (breaking) (minecraft:xxx_mushroom_block) [trees.allowedBreakingLeaves]

**Reply by Some-Doofus** [reply]
I got the same error even without the mushroom blocks in the allowed adjacent blocks list, I just tried that to see if it would fix it. I also tried putting the red/brown mushroom blocks under both the decaying leaves and non-decaying leaves lists (not simultaneously) and still got the error. Not sure what else would be causing the problem. I do have several other mods but none I would have thought would conflict in this way considering normal trees work fine.

**Reply by Rakambda** [reply]
If you still have the same error then it means that you have things defined in the adjacent blocks config, and they do not match. It seems like you don't really wanna use that feature and should therefore keep that list empty.

Example of a config file that works for breaking mushrooms:
```json
{
  "trees": {
    "allowedLogs": [
      "minecraft:mushroom_stem"
    ],
    "deniedLogs": [],
    "allowedLeaves": [],
    "allowedNonDecayLeaves": [
      "minecraft:red_mushroom_block",
      "minecraft:brown_mushroom_block"
    ],
    "deniedLeaves": [],
    "breakMode": "INSTANTANEOUS",
    "detectionMode": "WHOLE_TREE",
    "maxScanSize": 500,
    "maxSize": 100,
    "maxSizeAction": "ABORT",
    "breakOrder": "FURTHEST_FIRST",
    "minimumLeavesAroundRequired": 0,
    "includePersistentLeavesInRequiredCount": true,
    "treeBreaking": true,
    "leavesBreaking": true,
    "leavesBreakingForceRadius": 0,
    "allowMixedLogs": false,
    "breakNetherTreeWarts": true,
    "instantlyBreakWarts": false,
    "searchAreaRadius": -1,
    "allowedAdjacentBlocks": [],
    "adjacentStopMode": "STOP_ALL"
  },
  "tools": {
    "allowed": [],
    "denied": [],
    "preserve": false,
    "ignoreTools": false,
    "damageMultiplicand": 1.0,
    "damageRounding": "ROUND_DOWN",
    "speedMultiplicand": 0.0
  },
  "player": {
    "allowedTags": []
  },
  "enchantment": {
    "registerEnchant": false,
    "registerSpecificEnchant": false,
    "hideEnchant": false
  },
  "sneakMode": "SNEAK_DISABLE",
  "breakInCreative": true,
  "notificationMode": "ACTION_BAR"
}
```

**Reply by Some-Doofus** [reply]
Ok removing all entries from the adjacent list worked, but I wanted to have dirt and grass blocks in there at least for times when trees are not on flat ground and touch these blocks. Is there a reason having them whitelisted would be causing this error?

**Reply by Rakambda** [reply]
I'll improve the error message with #347 so that it displays the actual block name that wasn't in the adjacent list.
To me it just sounds like the tree is touching another block that wasn't in the adjacent list.

**Reply by Rakambda** [reply]
And it's probably `minecraft:air` that wasn't whitelisted

**Reply by Some-Doofus** [reply]
It seems that it only works if the list is empty. Even with only dirt or dirt and air added nothing will cut. And that's to say not even normal trees cut with anything in the adjacent blocks whitelist. I previously thought they weren't affected but didn't actually test till now

**Reply by Rakambda** [reply]
3.5.1 Is being released (maybe wait 5mins). maybe try with that version as it'll tell you the name of the block being adjacent and not whitelisted.

**Reply by Some-Doofus** [reply]
I guess I misunderstood the adjacent blocks whitelist anyway, with all blocks being whitelisted if it is left empty. I thought I had to manually define any and all blocks I wanted whitelisted or else any block would stop the cut. Still, this means there's no way to define blocks that will prevent the cut, but that's not much of a problem. Thanks anyway for your help.
